github/orgs_audit_log.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+// GetAuditLogOptions sets up optional parameters to query audit-log endpoint.
+type GetAuditLogOptions struct {
+	Phrase  *string `json:"phrase,omitempty"`   // A search phrase. (Optional.)
+	Include *string `json:"include,omitempty"`  // Event type includes. Can be one of "web", "git", "all". Default: "web". (Optional.)
+	After   *string `json:"after,omitempty"`    // A cursor, as given in the Link header. If specified, the query only searches for events after this cursor. (Optional.)
+	Before  *string `json:"before,omitempty"`   // A cursor, as given in the Link header. If specified, the query only searches for events before this cursor. (Optional.)
+	Order   *string `json:"order,omitempty"`    // The order of audit log events. Can be one of "asc" or "desc". Default: "desc". (Optional.)
+	PerPage *int64  `json:"per_page,omitempty"` // Results per page (max 100).
+}
+
+// AuditEntry describes the fields that may be represented by various audit-log "action" entries.
+type AuditEntry struct {
```

Comment:
I'm extremely confused at the moment.
When I look at the examples listed here:
https://docs.github.com/en/rest/reference/orgs#get-the-audit-log-for-an-organization

I see three audit entry examples, with just a tiny handful of fields.

Where is this huge list of fields coming from?

Reply:
We have an internal (not yet public) representation of the these exposed API fields for the 600+ events that the audit-log supports.  Some of these are defined here: https://docs.github.com/en/github/setting-up-and-managing-organizations-and-teams/reviewing-the-audit-log-for-your-organization#audit-log-actions (not all these actions will be available in an org/enterprise setting).

Reply:
Thinking out aloud should we rather have these be more tightly coupled to the Action and then do a lookup based on `action` or maybe embed other Actions event fields?

so instead of 
```
type AuditEntry struct {
     <smattering of fields>
} 
```
do something more akin to 
```
type AuditEntry struct{
   Action *string `json:"action,omitempty"`
    DocumentID *string `json:"_document_id,omitempty"`
    Timestamp *int64 `json:"*timestamp,omitempty"`

    GitEventAuditEntry
    TeamEventAuditEntry
}

type GitEventAuditEntry struct {
   TransportProtcol *string `json:"transport_protocol,omitempty"`
    ....
}
```

Reply:
Hmm... great question.  I'm fine with making one huge struct like we do for some of the other responses, but I also see the value of your idea...

I suppose we might want to consider how challenging one vs the other might be for maintenance of this repo as things change on the GitHub side.

How about you implement what makes sense to you, then we can see what it looks like and others can comment if they have an opinion as well.

Reply:
I went down this path of separating the fields into logical structs and decided against it only because I think discoverability of the fields might suffer and not all fields map 1:1 to an audit-action event. Instead I sorted the field list within here so that it's easily discoverable and gives the user a comprehensive look of what can be expected back.